supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesAsyncBackendJobExecutionActor.scala

```diff
@@ -311,12 +317,71 @@ object TesAsyncBackendJobExecutionActor {
         getTesStatusFn(Option(state), previousCostData, handle.pendingJob.jobId)
       }
     }
+
+  def onTaskComplete(runStatus: TesRunStatus,
+                     handle: StandardAsyncPendingExecutionHandle,
+                     getTaskLogsFn: StandardAsyncPendingExecutionHandle => Future[Option[TaskLog]],
+                     tellMetadataFn: Map[String, Any] => Unit,
+                     tellBardFn: StandardAsyncRunState => Unit,
+                     logger: LoggingAdapter
+  )(implicit ec: ExecutionContext): Unit = {
+    val logs = getTaskLogsFn(handle)
+    val taskEndTime = getTaskEndTime(logs)
+
+    val errors = for {
+      errors <- runStatus match {
+        case Error(_, _) | Failed(_, _) => getErrorSeq(logs)
+        case _ => Future.successful(Option(Seq.empty[String]))
+      }
+    } yield errors
+
+    errors.onComplete {
+      case Success(r) =>
+        if (r.nonEmpty) {
+          r.map(r => tellMetadataFn(Map(CallMetadataKeys.Failures -> r)))
+        }
+      case Failure(e) => logger.error(e.getMessage)
+    }
+
+    taskEndTime.onComplete {
+      case Success(result) =>
+        result.foreach(r => tellMetadataFn(Map(CallMetadataKeys.TaskEndTime -> r)))
+        val newCostData = runStatus.costData.map(_.copy(endTime = result))
+        runStatus match {
+          case _: Complete => tellBardFn(Complete(newCostData))
+          case _: Cancelled => tellBardFn(Cancelled(newCostData))
+          case failed: Failed => tellBardFn(Failed(sysLogs = failed.sysLogs, costData = newCostData))
+          case error: Error => tellBardFn(Error(sysLogs = error.sysLogs, costData = newCostData))
+          case _ => ()
```

Comment:
This onTaskComplete method is identical to before except for the new runStatus match and tellBard call